src/gluonts/dataset/split/splitter.py

```diff
@@ -224,8 +225,10 @@ def split(self, items: List[DataEntry]) -> TrainTestSplit:
             test = self._trim_history(self._test_slice(item))
 
             split._add_train_slice(train)
-
-            assert len(test) - len(train) >= getattr(self, "prediction_length")
+            if (getattr(self, "max_history")) is None:
+                assert len(test) - len(train) >= getattr(
+                    self, "prediction_length"
+                )
```

Comment:
Why is the guard needed here? Doesn't the condition need to hold in any case?

Reply:
As I understand it, the reason to use the max_split parameter is to shorten the test set.
For some models it will be enough to use `max_history = context_length+prediction_length`. In that case the test set has length `context_length+prediction_length`, while the train set might be much larger.

Reply:
> What always holds is something like end_date_train+pd.Timedelta(prediction_length) == end_date_test

Or maybe something like

`end_date_train + prediction_length * end_date_train.freq <= end_date_test`

Reply:
I think we should keep a meaningful assertion without any condition, because this is essentially checking whether it makes sense to do backtesting on the test data, for the given prediction length, given that you trained on this training data